examples/server/server.cpp

```diff
@@ -124,8 +124,9 @@ static void server_log(const char *level, const char *function, int line,
 static std::string tokens_to_output_formatted_string(const llama_context *ctx, const llama_token token)
 {
     std::string out = token == -1 ? "" : llama_token_to_str(ctx, token);
-    // if first bit is 1, meaning it's a partial character
-    if (out.size() > 0 && (out[0] & 0x80) == 0x80)
+    // if the size is 1 and first bit is 1, meaning it's a partial character
+    //   (size > 1 meaning it's already a known token)
+    if (out.size() == 1 && (out[0] & 0x80) == 0x80)
```

Comment:
I got a bug on `out.size() > 1`, if some token is a known vocab like `你`, it will return the first byte of the vocab.

After this fix, we can see the expected result instead of `byte: ...`:
<img width="435" alt="Screenshot 2023-08-21 at 11 44 32" src="https://github.com/ggerganov/llama.cpp/assets/3001525/2fd1109b-cad4-4d36-9793-1766200ab76d">